### PR TITLE
Refactor BOD 18-01 Lambda configuration

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request does a refactor in BOD 18-01 Lambda resources. This includes consolidating Lambda information into a single Terraform variable, preferring `for_each` to `count` for resource creation, and only creating resources if Lambda configurations have been provided.

## 💭 Motivation and context ##

This refactor does a bit of modernization of the Terraform for these Lambdas and adds the ability to deploy this configuration with no BOD 18-01 Lambda resource. Since the architecture that uses these Lambdas relies on a specific function naming scheme, and function names in an account are unique per region, this is a cleaner solution than alternate Lambda function names when multiple development environments are deployed in the same region.

## 🧪 Testing ##

Automated tests pass. I confirmed I could deploy this configuration both with and without BOD 18-01 Lambdas in my development environment.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Nice improvements! 🔥

File: terraform/README.md [hunk 1]
```diff
@@ -652,6 +652,8 @@ terraform apply -var-file=<your_workspace>.tfvars
 | assessment\_data\_s3\_bucket | The name of the bucket where the assessment data JSON file can be found.  Note that in production Terraform workspaces, the string '-production' will be appended to the bucket name.  In non-production workspaces, '-<workspace\_name>' will be appended to the bucket name. | `string` | `""` | no |
 | aws\_availability\_zone | The AWS availability zone to deploy into (e.g. a, b, c, etc.). | `string` | `"a"` | no |
 | aws\_region | The AWS region to deploy into (e.g. us-east-1). | `string` | `"us-east-1"` | no |
+| bod\_lambda\_function\_bucket | The name of the S3 bucket where the Lambda function zip files reside.  Terraform cannot access buckets that are not in the provider's region, so the region name will be appended to the bucket name to obtain the actual bucket where the zips are stored.  So if we are working in region us-west-1 and this variable has the value buckethead, then the zips will be looked for in the bucket buckethead-us-west-1. | `string` | n/a | yes |
```
Comment (jsf9k) [inline terraform/README.md]:
```suggestion
| bod\_lambda\_function\_bucket | The name of the S3 bucket where the Lambda function zip files reside.  Terraform cannot access buckets that are not in the provider's region, so the region name will be appended to the bucket name to obtain the actual bucket where the zips are stored.  So if we are working in region `us-west-1` and this variable has the value `buckethead`, then the zips will be looked for in the bucket `buckethead-us-west-1`. | `string` | n/a | yes |
```

File: terraform/variables.tf [hunk 2]
```diff
@@ -19,6 +19,11 @@ variable "assessment_data_import_lambda_s3_key" {
   type        = string
 }
 
+variable "bod_lambda_function_bucket" {
+  description = "The name of the S3 bucket where the Lambda function zip files reside.  Terraform cannot access buckets that are not in the provider's region, so the region name will be appended to the bucket name to obtain the actual bucket where the zips are stored.  So if we are working in region us-west-1 and this variable has the value buckethead, then the zips will be looked for in the bucket buckethead-us-west-1."
```
Comment (jsf9k) [inline terraform/variables.tf]:
```suggestion
  description = "The name of the S3 bucket where the Lambda function zip files reside.  Terraform cannot access buckets that are not in the provider's region, so the region name will be appended to the bucket name to obtain the actual bucket where the zips are stored.  So if we are working in region `us-west-1` and this variable has the value `buckethead`, then the zips will be looked for in the bucket `buckethead-us-west-1`."
```

**Review by jsf9k — APPROVED** [review]
I have a minor formatting suggestion, but I approve this pull request even if you choose not to accept it.

**Review by jasonodoom — APPROVED** [review]

**Pull request merged**